documentation/src/main/asciidoc/mapper-orm-bridge-index-field-dsl.asciidoc

```diff
@@ -165,3 +165,105 @@ and pass it to the bridge for later use.
 <6> Create sub-fields, get references to these fields
 and pass them to the bridge for later use.
 ====
+
+[[mapper-orm-bridge-index-field-dsl-dynamic]]
+== Dynamic fields with field templates
+
+Field declared in the sections above are all _static_:
+their path and type are known on bootstrap.
+
+In some very specific cases, the path of a field is not known until you actually index it;
+for example, you may want to index a `Map<String, Integer>` by using the map keys as field names.
+The field, then, is considered _dynamic_.
+
+Dynamic fields are not declared on bootstrap,
+but need to match a field _template_ that is declared on bootstrap.
+The template includes the field type and structural information (multi-valued or not, ...),
+but omits the field name.
```

Comment:
I would use the term names instead of name here, to make clear that for each template there could be more fields.